### PR TITLE
Update Python dependencies

### PR DESCRIPTION
**Pull request by galaxybot**
by running `make update-dependencies`.

### CODE REVIEW

File: lib/galaxy/dependencies/dev-requirements.txt [hunk 1]
```diff
@@ -171,7 +171,7 @@ trio==0.26.0 ; python_version >= "3.8" and python_version < "3.13"
 tuspy==1.0.3 ; python_version >= "3.8" and python_version < "3.13"
 twill==3.2.5 ; python_version >= "3.8" and python_version < "3.13"
 twine==5.1.1 ; python_version >= "3.8" and python_version < "3.13"
-typing-extensions==4.12.2 ; python_version >= "3.8" and python_version < "3.13"
+typing-extensions==4.9.0 ; python_version >= "3.8" and python_version < "3.13"
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/dev-requirements.txt]: https://github.com/SeleniumHQ/selenium/issues/14278 , should be fixed in selenium==4.23.1

File: lib/galaxy/schema/workflows.py [hunk 2]
```diff
@@ -194,19 +195,21 @@ class StoredWorkflowDetailed(StoredWorkflowSummary):
...
+        Annotated[
+            Union[
+                InputDataStep,
+                InputDataCollectionStep,
+                InputParameterStep,
+                PauseStep,
+                ToolStep,
+                SubworkflowStep,
+            ],
+            Field(discriminator="type"),
```
Comment (mvdbeek) [inline lib/galaxy/schema/workflows.py]: That's a good catch, we should backport that!

Comment (nsoranzo) [reply lib/galaxy/schema/workflows.py]: Absolutely, I'll open a separate PR as I am not sure when this one will be ready.

**Comment by nsoranzo** [comment]
Ready to review at last!
Test failures unrelated.

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**